scanner/src/main/kotlin/experimental/PackageProvenanceResolver.kt

```diff
@@ -134,16 +134,31 @@ class DefaultPackageProvenanceResolver(
             }
         }
 
-        val request = Request.Builder().head().url(pkg.sourceArtifact.url).build()
-        OkHttpClientHelper.execute(request).use { response ->
-            if (response.code == HttpURLConnection.HTTP_OK) {
-                val artifactProvenance = ArtifactProvenance(pkg.sourceArtifact)
-                storage.putProvenance(pkg.id, pkg.sourceArtifact, ResolvedArtifactProvenance(artifactProvenance))
-                return artifactProvenance
-            }
+        val responseCode = requestSourceArtifact(pkg, "HEAD").takeIf { it != HttpURLConnection.HTTP_BAD_METHOD }
```

Comment:
Nit: Maybe `takeUnless { it == HttpURLConnection.HTTP_BAD_METHOD }` reads a bit nicer?